src/utils/general.ts

```diff
@@ -891,11 +893,20 @@ export function waitForPageToBeVisible() {
   });
 }
 
+export function getBrowserCurrentLocale() {
```

Comment:
I am not sure if I want to use the complete browser locale for this.

I was planning to replace all the hour, mins etc with the Intl functions of the browser, so we do not have to do anything for the translation of time. But the problem I have is that I am not sure how to get the locale, the only way I could find is to creating an translation string for every language that contains the locale in it. A bit dirty but I do not see any cons in doing it like this. It would even correctly return it for the force english option. 

This would be a much better cache key indicator in this case and it would work well with the DurationFormat as locale too. What do you think about that?


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat

Reply:
To be honest I never touched this namespace, but looking into it, DurationFormat looks cool for durations I agree, but also [DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) I guess is nice, since US/Europe time formats.

But what you said about locale - aren't you already getting locale from browser with i18n inside `api.storage.lang`? I think it already checking whole browser lang when you call `chrome.i18n.getMessage`? 

And the reason I tried getting browser locale - to synchronize info and statistics fields inside meta overview when browser locale change. They need to be somehow handled, you can't just "update" them. the only thing I can think of is somehow evaluate their value on puling from local storage, or what I did. I don't know how to do the former one

Reply:
Ah I nearly forgot that DurationFormat is not supported in firefox. So we would need to implement an fallback for it until it is fully supported.

Reply:
> Ok let me show what I mean. I changed my browser lang from RU->ENG. The whole pop up UI lang also changed.

Ok I did get that. Probably just misinterpreted it and I do not mean not to use the locale. Just to use the malsync specif locale from the `chrome.i18n.getMessage` which would automatically fallback and support force english.

> No, I just assume they use the same function chrome.i18n.getUILanguage to get needed locale inside chrome.i18n.getMessage.  

I did not test it, but I imagine it returns the UI language of the browser, which can be different from the website prefered language (what the normal window js api returns) and again is different from the locale that malsync uses. But that is what I at least though. Never tested it.

> But they won't?... I don't get you, what part of the code you talking about? I changed some occurrences of "mins" "hours" etc, yes, but they pulled with api.storage.lang so they would be the same as UI. They already exist in messages.json.

Was probably not the best idea to mix the DurationFormat into the text like. I was talking about that and how you can pass the locale and that it would be a good key for the cache 

> You know, yes, I think it can solve getting locale. But one thing - how do you think it would work with Weblate? Can people mess with this string or you personally review translation PRs? If it's okay then I just add locale field in every locale file?

We review the changes. It opens an PR we have to accept.

> True just tested it. FF does have DateFormats btw. I can add them too if you ok?

Probably better in a seperate PR after this is done.

---

But genererally this is still in the discussion phase. So feel free to suggest alternatives if you have better. That is just what I was thinking about.

Reply:
>  it returns the UI language of the browser

True, this is browsers language. The preferred languages are in `navigator.language` (most preferred) and `navigator.languages` (all preferred)

> I was talking about that and how you can pass the locale and that it would be a good key for the cache

Got you, no problem

> discussion phase

To make it simpler for us, if you okay with this i just summarize:
1. Add locale translation key to every `message.json`
2. Pass it into Cache key parameter when it creates an entry
3. When getting overview meta from said cache, check current locale with cached one
3.1 If not equal - delete Cache entry and create new one
3.2 If equal - return it

I'm good with this as long as there are no "unexpected things" which I can't think of any right now.
And for DateFormat and DurationFormat I'll do another PR. How is it?

Reply:
Sounds good.

I think it would make sense to directly implement the durationformater, or at least directly use this utility for later. I think it pretty much does what durationformater does and could directly be used as fallback.

https://github.com/MALSync/MALSync/blob/master/src/utils/time.ts#L134